File: frontend/src/types/car.ts
export interface User {
  id: number;
  name: string;
  email: string;
  role: 'user' | 'dealer' | 'rental_agency' | 'admin';
  phone?: string;
  showroom_name?: string;
  city?: string;
  is_verified?: boolean;
}

export interface PriceRating {
  badge: 'great' | 'fair' | 'high';
  label: string;
  color: string;
  bg: string;
  icon: string;
}

export interface Car {
  id: number;
  user_id: number;
  brand: string;
  model: string;
  year: number;
  price: number;
  transaction_type: 'sale' | 'rent';
  description?: string;
  city: string;
  condition: 'new' | 'used';
  mileage?: number;
  transmission?: 'automatic' | 'manual';
  fuel_type?: 'petrol' | 'diesel' | 'hybrid' | 'electric';
  engine_power?: number;
  body_type?: 'suv' | 'sedan' | 'hatchback' | 'coupe' | 'truck' | 'van';
  status?: 'active' | 'sold' | 'reserved';
  image_url?: string;
  is_available: boolean;
  price_rating?: PriceRating;
  user?: User;
  created_at?: string;
  updated_at?: string;
}

export interface Rental {
  id: number;
  user_id: number;
  car_id: number;
  start_date: string;
  end_date: string;
  total_price: number;
  status: 'pending' | 'confirmed' | 'cancelled' | 'completed';
  car?: Car;
  created_at?: string;
}

export interface ReviewItem {
  id: number;
  user_id: number;
  dealer_id: number;
  rating: number;
  comment?: string;
  created_at: string;
  user?: {
    id: number;
    name: string;
  };
}

export interface Dealer {
  id: number;
  name: string;
  showroom_name: string;
  role: 'dealer' | 'rental_agency' | 'user';
  phone?: string;
  city?: string;
  is_verified?: boolean;
  cars_count: number;
  average_rating: number;
  total_reviews: number;
  cars?: Car[];
  reviews?: ReviewItem[];
}

File: mobile/lib/models/car.dart
class Car {
  final int id;
  final int userId;
  final String brand;
  final String model;
  final int year;
  final double price;
  final String transactionType;
  final String? description;
  final String city;
  final String condition;
  final String? imageUrl;
  final String? ownerName;

  Car({
    required this.id,
    required this.userId,
    required this.brand,
    required this.model,
    required this.year,
    required this.price,
    required this.transactionType,
    this.description,
    required this.city,
    required this.condition,
    this.imageUrl,
    this.ownerName,
  });

  // دالة تحويل الـ JSON القادم من Laravel API إلى كائن Car
  factory Car.fromJson(Map<String, dynamic> json) {
    return Car(
      id: json['id'],
      userId: json['user_id'] ?? 0,
      brand: json['brand'] ?? '',
      model: json['model'] ?? '',
      year: json['year'] ?? 0,
      price: double.tryParse(json['price'].toString()) ?? 0.0,
      transactionType: json['transaction_type'] ?? 'sale',
      description: json['description'],
      city: json['city'] ?? '',
      condition: json['condition'] ?? 'used',
      imageUrl: json['image_url'],
      ownerName: json['user'] != null ? json['user']['name'] : null,
    );
  }
}
